sdk/lib/atl/atlimage.h

```diff
@@ -93,28 +93,40 @@ class CImage
 
     HDC GetDC() const throw()
     {
-        if (m_hDC)
-            return m_hDC;
+        ATLASSERT(m_nDCRefCount >= 0);
+
+        if (::InterlockedIncrement(&m_nDCRefCount) == 1)
```

Comment:
Is that enough / what Wndows does?
It feels the whole block should be protected, not this increment only.

Reply:
I cannot see the original directly.
I'm not sure. Helper needed.